cast/src/lib.rs

```diff
@@ -1656,6 +1662,75 @@ impl SimpleCast {
     pub fn disassemble(bytecode: &str) -> Result<String> {
         format_operations(disassemble_str(bytecode)?)
     }
+
+    /// Gets the selector for a given function signature
+    /// Optimizes if the `optimize` parameter is set to a number of leading zeroes
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), None)?.0, String::from("0xbd0d639f"));
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), Some(2))?.0, String::from("0x0000bf3e"));
+    ///     assert_eq!(Cast::get_selector(String::from("foo(address,uint256)"), Some(2))?.1, Some(String::from("foo71661(address,uint256)")));
+    ///
+    ///     Ok(())
+    /// }
+    /// ```    
+    pub fn get_selector(
+        signature: String,
+        optimize: Option<usize>,
+    ) -> Result<(String, Option<String>, Option<Duration>)> {
+        if optimize.is_none() || optimize.unwrap() > 4 {
```

Comment:
Avoid unwrapping

Also it's probably a better API to just cap it at `4` instead of returning the default sig for values above

```suggestion
        let optimize = optimize.unwrap_or(0).max(4);
        if optimize == 0 {
```